subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/resolveengine/graph/builder/EdgeState.java

```diff
@@ -216,6 +216,11 @@ private void calculateTargetConfigurations(ComponentState targetComponent) {
                     for (EdgeState otherEdge : unattachedDependencies) {
                         if (otherEdge != this && !otherEdge.isConstraint()) {
                             otherEdge.attachToTargetConfigurations();
+                            if (otherEdge.targetNodeSelectionFailure != null) {
+                                // Copy selection failure
+                                this.targetNodeSelectionFailure = otherEdge.targetNodeSelectionFailure;
+                                return;
```

Comment:
For my understanding, I think not, but would it be possible to have more than one unattached dependency in this case?

Reply:
It should not, especially since I fixed a leak of those recently. Though from what I understand, what matters is that the constraint is connected. Once it happens, changes to target nodes at a later stage will be properly reflected on that edge as well.